.github/workflows/test-cpu.yml

```
@@ -0,0 +1,116 @@
+name: Python application
+
+on:
+  push:
+    branches:
+      - main
+      - "*.*.x"
+  pull_request:
+    branches:
+      - main
+      - "*.*.x"
+# Cancel the job if new commits are pushed
+# https://stackoverflow.com/questions/66335225/how-to-cancel-previous-runs-in-the-pr-when-you-push-new-commitsupdate-the-curre
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  pytest:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - python-version: 3.13
+            test-type: coverage
+          - python-version: 3.11
+            test-type: standard
+          - python-version: 3.13
+            dependencies-version: pre-release
+            test-type: strict-warning
+          - python-version: 3.11
+            dependencies-version: minimum
+            test-type: coverage
```

Better not to use floats for version numbers.

```suggestion
          - python-version: '3.13'
            test-type: coverage
          - python-version: '3.11'
            test-type: standard
          - python-version: '3.13'
            dependencies-version: pre-release
            test-type: strict-warning
          - python-version: '3.11'
            dependencies-version: minimum
            test-type: coverage
```